### ISSUE TITLE
State pruning problems in MS

### ISSUE THREAD

**Issue by palango** [raiden-network/raiden-services#1089]
## Problem Definition

```
ms_1                 | 2021-10-04 11:53:18.395330 [info     ] Triggering channel monitoring  [monitoring_service.handlers]
ms_1                 | 2021-10-04 11:53:18.793888 [error    ] Error during event handler     [monitoring_service.service] handled_event={'token_network_address': '0x625F82D937ccA0f1fF0097864895ba91635309A3', 'channel_identifier': 7129, 'non_closing_participant': '0x62259B40ea5A35EEa5bc996108F0712ebEF812F9', 'type_name': 'ActionMonitoringTriggeredEvent'}
ms_1                 | Traceback (most recent call last):
ms_1                 |   File "/services/src/monitoring_service/service.py", line 77, in handle_event
ms_1                 |     handler(event, context)
ms_1                 |   File "/services/src/monitoring_service/handlers.py", line 474, in action_monitoring_triggered_event_handler
ms_1                 |     to_block=latest_block,
ms_1                 |   File "/services/src/raiden_libs/blockchain.py", line 288, in get_pessimistic_udc_balance
ms_1                 |     for block in [from_block, to_block + 1]
ms_1                 |   File "/services/src/raiden_libs/blockchain.py", line 288, in <genexpr>
ms_1                 |     for block in [from_block, to_block + 1]
ms_1                 |   File "/usr/local/lib/python3.7/site-packages/web3/contract.py", line 968, in call
ms_1                 |     **self.kwargs
ms_1                 |   File "/usr/local/lib/python3.7/site-packages/web3/contract.py", line 1504, in call_contract_function
ms_1                 |     state_override=state_override,
ms_1                 |   File "/usr/local/lib/python3.7/site-packages/web3/module.py", line 60, in caller
ms_1                 |     null_result_formatters)
ms_1                 |   File "/usr/local/lib/python3.7/site-packages/web3/manager.py", line 190, in request_blocking
ms_1                 |     null_result_formatters)
ms_1                 |   File "/usr/local/lib/python3.7/site-packages/web3/manager.py", line 168, in formatted_response
ms_1                 |     raise ValueError(response["error"])
ms_1                 | ValueError: {'code': -32000, 'message': 'This request is not supported because your node is running with state pruning. Run with --pruning=archive.'}
```

**Comment by palango** [comment]
Got this after switching to geth:
```
ms_1                 | 2021-10-04 14:41:49.580961 [error    ] Error during event handler     [monitoring_service.service] handled_event={'token_network_address': '0x625F82D937ccA0f1fF0097864895ba91635309A3', 'channel_identifier': 7269, 'non_closing_participant': '0x045BB1588077b68AeeeA039a269Baa1020e92671', 'type_name': 'ActionMonitoringTriggeredEvent'}
ms_1                 | Traceback (most recent call last):
ms_1                 |   File "/services/src/monitoring_service/service.py", line 77, in handle_event
ms_1                 |     handler(event, context)
ms_1                 |   File "/services/src/monitoring_service/handlers.py", line 474, in action_monitoring_triggered_event_handler
ms_1                 |     to_block=latest_block,
ms_1                 |   File "/services/src/raiden_libs/blockchain.py", line 288, in get_pessimistic_udc_balance
ms_1                 |     for block in [from_block, to_block + 1]
ms_1                 |   File "/services/src/raiden_libs/blockchain.py", line 288, in <genexpr>
ms_1                 |     for block in [from_block, to_block + 1]
ms_1                 |   File "/usr/local/lib/python3.7/site-packages/web3/contract.py", line 968, in call
ms_1                 |     **self.kwargs
ms_1                 |   File "/usr/local/lib/python3.7/site-packages/web3/contract.py", line 1504, in call_contract_function
ms_1                 |     state_override=state_override,
ms_1                 |   File "/usr/local/lib/python3.7/site-packages/web3/module.py", line 60, in caller
ms_1                 |     null_result_formatters)
ms_1                 |   File "/usr/local/lib/python3.7/site-packages/web3/manager.py", line 190, in request_blocking
ms_1                 |     null_result_formatters)
ms_1                 |   File "/usr/local/lib/python3.7/site-packages/web3/manager.py", line 168, in formatted_response
ms_1                 |     raise ValueError(response["error"])
ms_1                 | ValueError: {'code': -32000, 'message': 'header not found'}

**Comment by palango** [comment]
The geth error might be fixed by https://github.com/raiden-network/raiden-services/pull/1090/commits/2ca88dfec87e0373fd48ceebcb3bbfdea4e42b90

**Comment by palango** [comment]
Fixed by #1090
